File: backend/VocabuAI.Api/Dtos/UserSettingsDto.cs
using VocabuAI.Application.Learning.Generation.Contracts;

namespace VocabuAI.Api.Dtos;

// Keep in sync with frontend/src/domain/dtos/UserSettingsDto.ts

public sealed record UserSettingsDto(
    string DefaultForeignFlashCardLanguage,
    string DefaultLocalFlashCardLanguage,
    bool HasOpenAiKey,
    string? OpenAiKeyLast4,
    int OpenAiMonthlyTokenLimit,
    int OpenAiTokensUsedThisMonth,
    string MonthKey,
    double? UsagePercent,
    AiProvider LastSelectedAiProvider
);

File: frontend/src/domain/dtos/UserSettingsDto.ts
// Keep in sync with backend/VocabuAI.Api/Dtos/UserSettingsDto.cs
import type { AiProvider } from "../AiProvider";

export type UserSettingsDto = {
  defaultForeignFlashCardLanguage: string;
  defaultLocalFlashCardLanguage: string;
  hasOpenAiKey: boolean;
  openAiKeyLast4: string | null;
  openAiMonthlyTokenLimit: number;
  openAiTokensUsedThisMonth: number;
  monthKey: string;
  usagePercent: number | null;
  lastSelectedAiProvider: AiProvider;
};
